cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -0,0 +1,172 @@
+        image_config = imagebuild.config.image_config
+        build_config = imagebuild.config.build_config
+        dev_settings_config = imagebuild.config.dev_settings_config
+
+        # TODO: use attributes from imagebuilder config instead of using these static variables.
+        core.CfnParameter(self, "EnableNvidia", type="String", default="false", description="EnableNvidia")
+        core.CfnParameter(self, "EnableDCV", type="String", default="false", description="EnableDCV")
+
+        # TODO: get archecture by ami id
+        architecture = "x86_64"
+        instance_types = ["p2.xlarge", "c5.xlarge"] if architecture == "x86_64" else ["m6g.xlarge"]
+
+        # Setup ImageBuilder Resources
+        # InstanceRole
+        if build_config.instance_role:
+            managed_policy_arns = [build_config.instance_role]
+        else:
+            managed_policy_arns = [
+                core.Fn.sub("arn:${AWS::Partition}:iam::aws:policy/AmazonSSMManagedInstanceCore"),
+                core.Fn.sub("arn:${AWS::Partition}:iam::aws:policy/EC2InstanceProfileForImageBuilder"),
+            ]
+        instancerole = iam.CfnRole(
+            self,
+            id="InstanceRole",
+            managed_policy_arns=managed_policy_arns,
+            assume_role_policy_document={
+                "Statement": {
+                    "Action": "sts:AssumeRole",
+                    "Effect": "Allow",
+                    "Principal": {"Service": "ec2.amazonaws.com"},
+                },
+                "Version": "2012-10-17",
+            },
+            path="/executionServiceEC2Role/",
+        )
+        instancerole.add_metadata("Comment", "Role to be used by instance during image build.")
+
+        # InstanceProfile
+        iam.CfnInstanceProfile(
+            self, id="InstanceProfile", path="/executionServiceEC2Role/", roles=[core.Fn.ref("InstanceRole")]
+        )
+
+        # InfrastructureConfiguration
+        imagebuilder.CfnInfrastructureConfiguration(
+            self,
+            id="PClusterImageInfrastructureConfiguration",
+            name="PCluster-Image-Infrastructure-Configuration",
+            instance_profile_name=core.Fn.ref("InstanceProfile"),
+            terminate_instance_on_failure=build_config.terminate_on_failure,
+            instance_types=instance_types,
+        )
+
+        # Define ami build instance ebs
+        ebs = imagebuilder.CfnImageRecipe.EbsInstanceBlockDeviceSpecificationProperty(
+            delete_on_termination=image_config.root_volume.delete_on_termination,
+            volume_size=image_config.root_volume.size,
+            volume_type="gp2",
+        )
+
+        cloudformation_dir = "../../../../cloudformation/imagebuilder/"
+
+        # Component
+        if dev_settings_config and dev_settings_config.update_and_reboot:
+            imagebuilder.CfnComponent(
+                self,
+                id="UpdateAndRebootComponent",
+                name="UpdateAndReboot",
```

Comment:
components names should be unique as well, we can use `generate_random_name_with_prefix`

Reply:
Done, I refactor `generate_random_name_with_prefix` and create a new function `generate_random_prefix`. So in the same stack, we can use same suffix/prefix for resource name.